packages/block-editor/src/store/selectors.js

```diff
@@ -2239,41 +2241,36 @@ export const __experimentalGetDirectInsertBlock = createSelector(
 	]
 );
 
-export const __experimentalGetParsedPattern = createSelector(
-	( state, patternName ) => {
-		const patterns = getAllPatterns( state );
-		const pattern = patterns.find( ( { name } ) => name === patternName );
-		if ( ! pattern ) {
-			return null;
-		}
-		return {
-			...pattern,
-			blocks: parse( pattern.content, {
-				__unstableSkipMigrationLogs: true,
-			} ),
-		};
-	},
-	( state ) => [ getAllPatterns( state ) ]
-);
-
-const getAllAllowedPatterns = createSelector(
-	( state ) => {
-		const patterns = getAllPatterns( state );
-		const { allowedBlockTypes } = getSettings( state );
-
-		const parsedPatterns = patterns
-			.filter( ( { inserter = true } ) => !! inserter )
-			.map( ( { name } ) =>
-				__experimentalGetParsedPattern( state, name )
+export const __experimentalGetParsedPattern = createRegistrySelector(
+	( select ) =>
+		createSelector( ( state, patternName ) => {
+			const { getAllPatterns } = unlock( select( STORE_NAME ) );
+			const patterns = getAllPatterns();
+			const pattern = patterns.find(
+				( { name } ) => name === patternName
 			);
-		const allowedPatterns = parsedPatterns.filter( ( { blocks } ) =>
-			checkAllowListRecursive( blocks, allowedBlockTypes )
-		);
-		return allowedPatterns;
-	},
-	( state ) => [ getAllPatterns( state ), state.settings.allowedBlockTypes ]
+			if ( ! pattern ) {
+				return null;
+			}
+			return {
+				...pattern,
+				blocks: parse( pattern.content, {
```

Comment:
This is another thing to think about: with lazy block loading, the `parse` function becomes async and `__exGetParsedPattern` must become a selector with resolver and associated reducer state. And all the callers, like `_exGetAllowedPatterns`, must be modified accordingly. I had it working in the #51778, but it deteriorated over time as pattern code in `trunk` got modified. And this PR will force another update.

Reply:
Is it a blocker for this PR? Should I adjust anything?

Reply:
Not a blocker, just something to consider -- that we'll probably want to evolve in that direction in near future. Is this PR going in the same direction (making that change easier) or in the opposite direction (making it harder)?

Reply:
Not sure it makes it easier or harder, it's probably the same as before. The only difference is that you can now take advantage of resolvers, so that might help.